extensions/amp-story/0.1/amp-story.js

```diff
@@ -269,8 +278,40 @@ export class AmpStory extends AMP.BaseElement {
 
 
   /** @private */
+  isAmpStoryEnabled_() {
+    if (isExperimentOn(this.win, TAG) || getMode().test || getMode().localDev) {
+      return true;
+    }
+
+    const origin = getSourceOrigin(this.win.location);
+    const hostName = parseUrl(origin).hostname;
+    const domains = hostName.split('.');
+
+    // Check all permutations of the domain to see if any level of the domain is
+    // whitelisted.  Taking the example of the whitelisted domain
+    // example.co.uk, if the page is served from www.example.co.uk/page.html:
+    //
+    //   www.example.co.uk => false
+    //   example.co.uk => true
+    //   co.uk => false
+    //   uk => false
+    //
+    // This is necessary, since we don't have any guarantees of which level of
+    // the domain is whitelisted.  For many domains (e.g. .com), the second
+    // level of the domain is likely to be whitelisted, whereas for others
+    // (e.g. .co.uk) the third level may be whitelisted.  Additionally, this
+    // allows subdomains to be whitelisted individually.
```

Comment:
Won't this also allow `example.co.uk.evil.com`?

Reply:
No, we check from each level of the domain to the end of the URL, so you'd get:

`example.co.uk.evil.com => false`
`co.uk.evil.com => false`
`uk.evil.com => false`
`evil.com => false`
`com => false`

Reply:
It *would* allow something like `evil.example.co.uk`:

`evil.example.co.uk => false`
`example.co.uk => true`
`co.uk => false`
`uk => false`

So, we need to make sure the domains we're whitelisting are in complete control of the authorities we're trying to whitelist.  AKA, whitelisting a generic host like `amazonaws.com` or `godaddy.com` is a bad idea.

Reply:
Simply put, this whitelists a domain and all of its subdomains.

Reply:
Gotcha, thanks.